src/JustSaying/Messaging/MessageSerialization/SystemTextJsonMessageBodySerializerBase.cs

```diff
@@ -0,0 +1,29 @@
+namespace JustSaying.Messaging.MessageSerialization;
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+/// <summary>
+/// Provides a base class for System.Text.Json-based message body serializers with default serialization options.
+/// </summary>
+public abstract class SystemTextJsonMessageBodySerializerBase
```

Comment:
I need to inherit it from a public class, this base class is just so that the static member isn't stored per generic `T`

Reply:
I'm still not following this. We're exposing a public class to consumers of the library that has no members - why not just make the member a `protected` property instead of a field?

Reply:
the field is `private protected`, which is a weird way of saying "it can be used by inherited types, but only from internal to the assembly".
The intent here was that `DefaultJsonSerializerOptions` is a private static field of `SystemTextJsonMessageBodySerializer<T>`, however static in a generic type means "static per generic T", which isn't what I want (and gives an analyzer warning).
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/private-protected

Reply:
That's what I mean - if I'm the consumer of JustSaying, it's an apparently empty public type, which is a bit weird.

Reply:
It feels like a simpler solution would be to have an `internal static class SystemTextJsonSerializerDefaults` to hold the default options, rather than having them in the inheritance chain. From what I can see, we only have the one deriving type, so I don't see the issue with losing the 1 per derived type relationship.